### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by ash2k**

### CODE REVIEW

File: cmd/server.go [hunk 1]
```diff
@@ -50,15 +50,12 @@ func parseRemoteAddr(addr string) string {
...
+func (s *Server) getRole(IP string) (string, error) {
 	var role string
 	var err error
 	operation := func() error {
 		role, err = s.store.Get(IP)
-		if err != nil {
-			return err
-		}
-		return nil
+		return err
```
Comment (ash2k) [inline cmd/server.go]: Simplified a bit

File: cmd/server.go [hunk 2]
```diff
@@ -74,6 +71,7 @@ func (s *Server) securityCredentialsHandler(w http.ResponseWriter, r *http.Reque
 	role, err := s.getRole(remoteIP)
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusNotFound)
+		return
```
Comment (ash2k) [inline cmd/server.go]: Missing returns

File: cmd/store.go [hunk 3]
```diff
@@ -9,9 +9,9 @@ import (
 
 // store implements the k8s framework ResourceEventHandler interface.
 type store struct {
-	iamRoleKey  string
-	mutex       sync.RWMutex
-	rolesByIP   map[string]string
+	iamRoleKey string
+	mutex      sync.RWMutex
+	rolesByIP  map[string]string
```
Comment (ash2k) [inline cmd/store.go]: `go fmt`

File: Makefile [hunk 4]
```diff
@@ -73,7 +73,10 @@ watch:
 cross:
 	CGO_ENABLED=0 GOOS=linux go build -o build/bin/linux/$(BINARY_NAME) $(GOBUILD_VERSION_ARGS) -a -installsuffix cgo  github.com/jtblin/$(BINARY_NAME)
 
-docker: cross
+cross-docker:
+	docker run --rm -v "$(GOPATH)":/opt/gopath -w /opt/gopath/src/github.com/jtblin/$(BINARY_NAME) -e GOPATH=/opt/gopath golang:latest go build -o build/bin/linux/$(BINARY_NAME) $(GOBUILD_VERSION_ARGS) -a -installsuffix cgo github.com/jtblin/$(BINARY_NAME)
```
Comment (ash2k) [inline Makefile]: I wanted to use `-race` to build it but got an error:
```
go build: -race requires cgo; enable cgo by setting CGO_ENABLED=1
```
After enabling it:
```
CGO_ENABLED=1 GOOS=linux go build -race -o build/bin/linux/kube2iam -ldflags "-s -X ""github.com/jtblin"/kube2iam"/version.Version=$(git describe --abbrev=0 --tags) -X ""github.com/jtblin"/kube2iam"/version.GitCommit=$(git rev-parse --short HEAD) -X ""github.com/jtblin"/kube2iam"/version.BuildDate=$(date +%Y-%m-%d-%H:%M)" -a -installsuffix cgo  github.com/jtblin/kube2iam
# runtime/cgo
ld: unknown option: --build-id=none
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [cross] Error 2
```

With this change the build happens in a docker container `golang:latest` i.e. in Linux environment. No need to have crazy cross-compilation setup locally. I think it is more reliable to use linux toolchain than using OS X compiler toolchain to build for linux.

Comment (jtblin) [inline Makefile]: I don't have any problem running `go build -trace .`.

**Comment by jtblin** [comment]
Would you mind keeping only the fixes in this PR and have a better commit message please?

File: Makefile (hunk 4 above)
Comment (ash2k) [inline Makefile]: Try adding `-race` (not `-trace`) to the `cross` make target.

Comment (ash2k) [inline Makefile]: Interestingly the size is almost the same. I expected a bigger difference:
```
make cross
ls -la ./build/bin/linux/kube2iam
....  25642176  8 Jun 15:03 ./build/bin/linux/kube2iam

make cross-docker
ls -la ./build/bin/linux/kube2iam
.... 25674376  8 Jun 15:05 ./build/bin/linux/kube2iam
```

**Comment by ash2k** [comment]
I removed the cross-compilation commit.

**Comment by jtblin** [comment]
Thanks a ton 👍

**Comment by jtblin** [comment]
Released `0.0.4`, see https://hub.docker.com/r/jtblin/kube2iam/tags/

**Pull request merged**